pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions_test.go

```diff
@@ -249,18 +271,33 @@ func TestAccessModeConflicts(t *testing.T) {
 		},
 	}
 
+	cmpOpts := []cmp.Option{
+		cmp.Comparer(func(s1 *framework.Status, s2 *framework.Status) bool {
```

Comment:
`framework.Status` already implements `Equal`. You shouldn't need this.

Reply:
I was encountering issues comparing statuses without this -- it gets hung-up on the pointer addresses:
```
--- FAIL: TestAccessModeConflicts (0.50s)
    --- FAIL: TestAccessModeConflicts/failed_to_get_PVC (0.10s)
        volume_restrictions_test.go:362: Unexpected Filter status (-want, +got):   (*framework.Status)(
            -   &⟪0xc0008f5a00⟫{
            -           code:    1,
            -           reasons: []string(⟪ptr:0xc000989890, len:1, cap:1⟫{`reading "PreFilterVolumeRestrictions" from cycleState: not found`}),
            -           err: &⟪0xc0008a6200⟫"fmt".wrapError{
            -                   msg: `reading "PreFilterVolumeRestrictions" from cycleState: not found`,
            -                   err: &⟪0xc0008181b0⟫"errors".errorString{s: "not found"},
            -           },
            -   },
            +   &⟪0xc0008f5e80⟫{
            +           code:    1,
            +           reasons: []string(⟪ptr:0xc000989d00, len:1, cap:1⟫{`reading "PreFilterVolumeRestrictions" from cycleState: not found`}),
            +           err: &⟪0xc0008a6420⟫"fmt".wrapError{
            +                   msg: `reading "PreFilterVolumeRestrictions" from cycleState: not found`,
            +                   err: &⟪0xc0008181b0⟫"errors".errorString{s: "not found"},
            +           },
            +   },
              )
```
I borrowed this pattern from here:
https://github.com/kubernetes/kubernetes/blob/v1.26.0/pkg/scheduler/schedule_one_test.go#L1484-L1488

Reply:
Can we compare the error log directly in `status.Equal()`?  Most of the time, we have a plain error.

Reply:
Updated comparator to use `.Message()` and explicitly set test fields `preFilterWantStatus` and `wantStatus` to `nil` to clarify test expectations